File: frontend/src/types/skillFile.ts
export interface SkillFile {
  id: number;
  user_id: number;
  name: string;
  description?: string | null;
  content: string;
  source: 'manual' | 'github' | 'marketplace';
  github_source?: { repo: string; path: string; branch: string; commit_sha?: string; raw_url: string } | null;
  status: 'draft' | 'published';
  version: number;
  created_at: string;
  updated_at: string | null;
}

export interface SkillFileCreateData {
  name: string;
  description?: string;
  content: string;
  source?: string;
  github_source?: unknown;
}

export interface SkillFileUpdateData {
  name?: string;
  description?: string;
  content?: string;
  status?: string;
}

export interface SkillFileMarketplaceItem {
  id: number;
  name: string;
  description?: string | null;
  content: string;
  source: string;
  author_id: number;
  author_name?: string;
  usage_count: number;
  created_at: string;
}

File: backend/app/schemas/skill_file.py
from pydantic import BaseModel
from datetime import datetime


class SkillFileBase(BaseModel):
    name: str
    description: str | None = None
    content: str


class SkillFileCreate(SkillFileBase):
    source: str = "manual"
    github_source: dict | None = None


class SkillFileUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    content: str | None = None
    status: str | None = None


class SkillFileResponse(SkillFileBase):
    id: int
    user_id: int
    source: str
    github_source: dict | None
    status: str
    version: int
    created_at: datetime
    updated_at: datetime | None

    class Config:
        from_attributes = True


class MountSkillRequest(BaseModel):
    skill_file_id: int
